### PR TITLE
feat(npm/react): allow custom webpack config path for react-scripts plugin

### LINKED ISSUE #16480: Component tests. plugins/react-scripts. Allow configure of webpack config path
### What would you like?
make it configurable as in `plugins/load-webpack` via `webpackFilename` 

### Why is this needed?
In my case config file is stored in different place than `react-scripts/config/webpack.config` so plugin doesn't work
before version 7 it was implemented in `load-webpack` plugin by adding cypress folder to transpile via options #14601

### PR DESCRIPTION
**Pull request by pashidlos**
- Closes #16480

### User facing changelog
`react-scripts` plugin now could be configured with custom webpack config path

```js
const injectDevServer = require('@cypress/react/plugins/react-scripts')
const path = require('path')

module.exports = (on, config) => {
  injectDevServer(
    on, 
    config, { 
      webpackConfigPath: './your/custom/path/webpack.config.js'  // <- added this option
    })

  return config
}

```

### PR Tasks
- [ ] Has the original issue or this PR been tagged with a release in ZenHub?

### CODE REVIEW

**Review by lmiller1990 — COMMENTED** [review]
Code seems generally fine. Two comments:

- have you tested this? I see we have not actually added a test. I wonder if we could add a simple unit test, perhaps? I can probably look into this if you aren't sure how or don't have the time. You'd just write it with mocha - two use cases would be loading w/ default location, and loading w/ a custom location.
- if you want this to go out asap, it'll need to target *master*. Otherwise it will need to wait for the next release (every 2 weeks).

Please let me know

1. do you plan to retarget for master, or happy to wait for 2 weeks?
2. about adding a unit test - can you do this, or do you need me to look into this?
3. have you actually tested this in some capacity, or should I do a manually QA on top of the unit test we should add before merging?

Thanks!

**Comment by pashidlos** [comment]
@lmiller1990 thanks for quick response 

> * do you plan to retarget for master, or happy to wait for 2 weeks?

retargeted to `master`, unfortunately passing pipeline is a challenge due to unstable tests

> * about adding a unit test - can you do this, or do you need me to look into this?

I wasn't able to find any example of those (should I add new npm script?)
please, assist with unit tests

> * have you actually tested this in some capacity, or should I do a manually QA on top of the unit test we should add before merging?

I checked it manually and the default behaviour is covered with `examples` projects AFAIK

**Review by lmiller1990 — APPROVED** [review]

**Pull request merged**
